SofaImGui/src/SofaImGui/ImGuiGUIEngine.cpp

```diff
@@ -1412,6 +1413,66 @@ void ImGuiGUIEngine::showComponents(const char* const& windowNameComponents, boo
                             ImGui::BulletText(t.c_str());
                         }
                     }
+
+                    ImGui::Separator();
+
+                    struct DataInfo
+                    {
+                        sofa::type::vector<std::string> templateType;
+                        std::string description;
+                        std::string defaultValue;
+                        std::string type;
+                    };
+
+                    std::map<std::string, std::map<std::string, DataInfo>> allData;
+                    {
+                        const auto tmpNode = core::objectmodel::New<simulation::graph::DAGNode>("tmp");
+                        for (const auto& [templateInstance, creator] : selectedEntry->creatorMap)
+                        {
+                            core::objectmodel::BaseObjectDescription desc;
+                            const auto object = creator->createInstance(tmpNode.get(), &desc);
```

Comment:
Everytime I see this I feel bad of instance of an object to get its datafield.

Reply:
Yes, but none really appeals me neither easy to implement.

The hard version is that I think this is connected on refactoring the way data are stored in object. Currently they are attributes of instances of sofa object while we are considered they are more attribute per class. This kind of refactoring in a discussion dating to years ago (with M. Nesme) in which it was pointing that rebuilding all the Data() everytime we instantiate an object was very time consume (and it is) which has performance cost for scene creates a lot of component dynamically.

As this is far too complex, I think the one approach worth exploring is to store the object detailed description encoded in an attribute of the factory classes entries. This attribute could be a json string. This description could be written manually and set when registering the object in the factory (either by passing a string or a filename containing the description) and if none provided, generated it by calling the creator we are just registering in the factory.